build-glibc.sh

```diff
@@ -488,6 +488,32 @@ if [ $DO_NATIVE_GDB = yes ]; then
     make_target building
     make_target_ordered installing install DESTDIR=$SYSROOTDIR
 
+    # libgmp
+    # Starting from GDB 11.1, it requires libgmp to be built natively [1].
+    #
+    # [1] gdb: Make GMP a required dependency for buidling GDB
+    # https://sourceware.org/git/?p=binutils-gdb.git;a=commit;h=1b4ac058f7d
+    gmp_version=6.1.2
+    gmp_url_base=https://gmplib.org/download/gmp
+    gmp_tar=gmp-${gmp_version}.tar.xz
+    gmp_url=$gmp_url_base/$gmp_tar
+    mkdir -p $toolchain_build_dir/_download_tmp
+    cd $toolchain_build_dir/_download_tmp
+    if [ ! -s $gmp_tar ]; then
+	$WGET -O $gmp_tar $gmp_url
+    fi
+
+    build_dir_init gmp
+    tar xf $toolchain_build_dir/_download_tmp/$gmp_tar --strip-components=1
+
+    configure_for_arc . $triplet \
+	--prefix=/usr \
+	--sysconfdir=/etc \
+	--program-prefix= \
+	--enable-static
+    make_target building
+    make_target_ordered installing install DESTDIR=$SYSROOTDIR
```

Comment:
Common code should be extracted into arc-init.sh , for example see a `build_expat` function shared by all targets: https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/blob/arc-releases/arc-init.sh#L527

Reply:
Done. As a bonus, I've also moved the `ncurses` part.